src/components/field/index.js

```diff
@@ -453,15 +449,9 @@ export default class TextField extends PureComponent {
   inputProps() {
     let store = {};
 
-    for (let key in TextInput.propTypes) {
-      if ('defaultValue' === key) {
-        continue;
-      }
-
-      if (key in this.props) {
-        store[key] = this.props[key];
-      }
-    }
+    Object.keys(this.props).forEach((key) => {
+      store[key] = this.props[key];
+    });
 
     return store;
   }
```

Comment:
Why not just return `this.props` at this point? Also unsure if this could trigger warnings / unexpected behaviour if arbitrary props get forwarded to the `TextInput` component below.

How about adding a whitelist with the supported `TextInput` props and filtering that?

Reply:
Yeah, both good points!

> Why not just return this.props at this point?

I opted to retain the loop to indicate where an escape clause for a prop could be easily added in the future, if the previously existing escape clause needed to be added again (e.g. `"defaultValue" === key`).

> Also unsure if this could trigger warnings / unexpected behaviour if arbitrary props get forwarded to the TextInput component below.

True - an oversight on my part as I'm a TypeScript user!

> How about adding a whitelist with the supported TextInput props and using it to filter the props?

I thought about this and I wasn't sure if the chance of an error in missing out a supported prop (either by mistake or because `react-native` adds one in the future) was worth the reward. Also, conscious of React Native not supporting prop types any more.

Happy for suggestions on what you think the best approach for this would be if you can think of an alternative to supporting a copied array of supported props.

Reply:
Yeah, I'm a TypeScript user myself, so I'm also all in favour of removing the `propTypes` overhead. The issue however is that the outer component accepts more props than a standard `TextInput`. Therefore, TypeScript should also accept these additional parameters without emitting a warning either.

The original uses `TextInput.propTypes` as a whitelist to filter out the `TextInput` props out from the full `TextField` props, so that nothing unexpected ends up there. Perhaps a blacklist would be better suited in the future - we already know which props are meant for the outer `TextField` component (e.g. by checking the `propTypes` defined above), so removing them should leave us with the props meant for the `TextInput`.